php/class-post-type.php

```diff
@@ -928,15 +963,24 @@ public function handle_snapshot_merge( $redirect_to, $do_action, $post_ids ) {
 	/**
 	 * Merge two or more snapshots
 	 *
-	 * @param array $post_ids post id array.
+	 * @param array  $posts  post array.
+	 * @param string $status Merged post status.
 	 *
 	 * @return int Changeset post id.
 	 */
-	public function merge_snapshots( $post_ids ) {
-		$posts = array_map( 'get_post', $post_ids );
+	public function merge_snapshots( $posts, $status = 'draft' ) {
 		usort( $posts, function( $a, $b ) {
-			$compare_a = $a->post_modified;
-			$compare_b = $b->post_modified;
+			$compare_key_a = 'post_modified';
+			$compare_key_b = 'post_modified';
+			// If status is scheduled then post_date takes more priority.
+			if ( get_post_status( $a ) === 'future' ) {
+				$compare_key_a = 'post_date';
+			}
+			if ( get_post_status( $b ) === 'future' ) {
+				$compare_key_b = 'post_date';
+			}
+			$compare_a = $a->$compare_key_a;
+			$compare_b = $b->$compare_key_b;
```

Comment:
@westonruter When you review this PR, I've made this change on snapshot merge function. 
This will give priority to scheduled post on merge. This was needed for preview future site state but this will also now affect changeset merge option.

Let know your thoughts, earlier we were merging based on `post_modified` without checking status.
Now in case of scheduled post it uses `post_date` otherwise it will use `post_modified`.

Reply:
This makes sense to me 👍